src/frontend/src/views/Upload/components/Samples/utils.ts

```diff
@@ -129,27 +129,49 @@ function handleFastaText(text: string, filename: string): ParseOutcome {
       continue;
     }
 
-    // Aggregate sequence line by line
-    for (let j = i + 1; j < lines.length; j++) {
-      const jLine = lines[j];
+    const { newIndex, sequence } = aggregateSequence(i, lines);
 
-      if (!jLine || jLine.includes(">")) {
-        i = j;
-        break;
-      }
+    i = newIndex;
 
+    if (sequence) {
       const id = iLine.replace(">", "");
 
       result = {
         ...result,
-        [id]: (result[id] || "") + jLine,
+        [id]: sequence,
       };
     }
   }
 
   return { errors, result };
 }
 
+function aggregateSequence(
+  i: number,
+  lines: string[]
+): { newIndex: number; sequence: string } {
+  let newIndex = i;
+  let sequence = "";
+
+  for (let j = i + 1; j < lines.length; j++) {
```

Comment:
can you add a quick comment on why we need` j=i+1` here? Is that to move past the sample id?

Reply:
That's right! Just added a comment to capture this 😄